src/context/card-context.js

```diff
@@ -1,67 +1,63 @@
-import React, {useState} from 'react';
+import React, { useState, useCallback, useEffect } from 'react';
 
 const CardCtx = React.createContext({
-    cards: [],
-    viewModeOnlyChecked: false,
-    show: false,
-    cardsCount: 0,
-    className: '',
-    getCount: () => {},
-    onAddCard: (newCard) => {},
-    onDeleteSelected: () => {},
-    setViewModeOnlyHandler: () => {},
-    cardUpdateHandler: (updatedCard) => {},
-    updateCheckedCardListHandler: (cardId, isChecked) => {},
-    showModal: () => {}
+  cards: [],
+  viewModeOnlyChecked: false,
+  show: false,
+  error: null,
+  cardsCount: 0,
+  className: '',
+  getCount: () => { },
+  onAddCard: (newCard) => { },
+  onDeleteSelected: () => { },
+  setViewModeOnlyHandler: () => { },
+  cardUpdateHandler: (updatedCard) => { },
+  updateCheckedCardListHandler: (cardId, isChecked) => { },
+  showModal: () => { }
 });
 
 export const CardCtxProvider = props => {
 
-const [viewModeOnlyChecked, setViewModeOnlyChecked] = useState(false);
-const [show, setShow] = useState(false);
-
-  const [cards, setCards] = useState([{
-    id: '1',
-    title: 'Core Java Volume I – Fundamentals',
-    description: 'Java reference book. ',
-  },
-  {
-    id: '2',
-    title: 'Title2',
-    description: 'Description2. ',
-  },
-  {
-    id: '3',
-    title: 'Title3',
-    description: 'Description3. ',
-  },
-  {
-    id: '4',
-    title: 'Title4',
-    description: 'Description4. ',
-  },
-  {
-    id: '5',
-    title: 'Title5',
-    description: 'Description5. ',
-  },
-  {
-    id: '6',
-    title: 'Title6',
-    description: 'Description6. ',
-  },
-  {
-    id: '7',
-    title: 'Title7',
-    description: 'Description7. ',
-  }
-  ]);
+  const [viewModeOnlyChecked, setViewModeOnlyChecked] = useState(false);
+  const [show, setShow] = useState(false);
+  const [error, setError] = useState(null);
+  const [cards, setCards] = useState([]);
+  const axios = require('axios');
+
+  const fetchCardsHandler = useCallback(async () => {
+    setError(null);
+    try {
+      const response = await axios.get('https://raw.githubusercontent.com/BrunnerLivio/PokemonDataGraber/master/output.json');
+      if (response.status !== 200) {
+        throw new Error("There is an issue...");
+      }
+      const data = await response.data;
+      let transformedData = [];
+      for (let i = 0; i <= data.length && i < 15; i++) {
```

Comment:
Do.Not.Use.for.For.Arrays.Ever. Only  FOREACH!!! Only HARDCORE!!!
Especially for limiting the data - really? SPLICE!!!!! MAP!!!!

Reply:
A-TA-TA!!!